apps/test/unit/blockly/cdoSerializationHelpersTest.js

```diff
@@ -0,0 +1,179 @@
+import {expect} from '../../util/reconfiguredChai';
+import {
+  addPositionsToState,
+  getNewLocation,
+  getCursorYAdjustment,
+  isBlockLocationUnset,
+} from '@cdo/apps/blockly/addons/cdoSerializationHelpers';
+
+describe('addPositionsToState', () => {
+  it('should add x/y values from XML to JSON serialization', () => {
+    const xmlBlocks = [
+      {
+        blockly_block: {
+          id: 'block1',
+          type: 'when_run',
+        },
+        x: 20,
+        y: 20,
+      },
+      {
+        blockly_block: {
+          id: 'block2',
+          type: 'flappy_whenCollideGround',
+        },
+        x: 230,
+        y: 20,
+      },
+      {
+        blockly_block: {
+          id: 'block3',
+          type: 'flappy_whenClick',
+        },
+        x: 20,
+        y: 140,
+      },
+    ];
+
+    const blockIdMap = new Map([
+      ['block1', {type: 'when_run', id: 'block1', x: 0, y: 0}],
+      ['block2', {type: 'flappy_whenCollideGround', id: 'block2', x: 0, y: 0}],
+      ['block3', {type: 'flappy_whenClick', id: 'block3', x: 0, y: 0}],
+    ]);
+
+    addPositionsToState(xmlBlocks, blockIdMap);
+
+    expect(blockIdMap.get('block1')).to.deep.equal({
+      type: 'when_run',
+      id: 'block1',
+      x: 20,
+      y: 20,
+    });
+    expect(blockIdMap.get('block2')).to.deep.equal({
+      type: 'flappy_whenCollideGround',
+      id: 'block2',
+      x: 230,
+      y: 20,
+    });
+    expect(blockIdMap.get('block3')).to.deep.equal({
+      type: 'flappy_whenClick',
+      id: 'block3',
+      x: 20,
+      y: 140,
+    });
+  });
+});
+
+describe('getNewLocation', () => {
+  it('should determine the new location based on the cursor', () => {
+    // Block and cursor details
+    const block = {id: 'block1', type: 'when_run', workspace: {RTL: false}};
+    const cursor = {x: 16, y: 16};
+
+    const newLocation = getNewLocation(block, cursor);
+
+    expect(newLocation).to.deep.equal({x: 16, y: 16});
+  });
+  it('should determine the new location for blocks with SVG frames', () => {
+    // Block and cursor details
+    const block = {
+      id: 'block4',
```

Comment:
Nit: having properties numbered like "block4" kind of implies to me that they mean something as the reader, which they don't in this context yeah? I think maybe you're trying to show "this is what a block object looks like", but I think it also can make your tests longer / harder to parse than they need to be.

If you want to show the default block structure, I like the style where you create some object like `defaultBlock` then create your test objects like `const block = {...defaultBlock, unusedSvg_: {}}`, which makes super clear what the properties are that are relevant in this test.

Reply:
Thanks for this feedback. There are definitely some block props that I had included that are more for illustrative purposes than anything and don't mean anything in the context of the tests. I decided to remove these so that it's easier to understand what we're actually testing. If a property is there, now it means it's required/assumed-to-exist by the function being tested.